components/bl-treeView-component/src/index.js

```diff
@@ -0,0 +1,138 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function TreeView({ component, eventHandlers }) {
+  const { display, style, classList, unprocessedTree, space } = component;
+  const { onClick } = eventHandlers;
+
+  const [itemsTree, setItemsTree] = useState([]);
+  const [parentItems, setParentItems] = useState([]);
```

Comment:
itemsList isn't better?

Don't clearly understand why it called parent items if it collection of all items.

Reply:
`parentItems` stores all items that have children

Reply:
for example, to be able to close or open all "folders" with an action without using recursion, or for possible other cases where some action with "folders" might be needed